src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs

```diff
@@ -1453,9 +1451,7 @@ public static Rune ToUpperInvariant(Rune value)
 
             if (GlobalizationMode.Invariant)
             {
-                // If the value isn't ASCII and if the globalization tables aren't available,
-                // case changing has no effect.
-                return value;
+                return UnsafeCreate(value.IsBmp ? CharUnicodeInfo.ToUpper((char) value._value) : CharUnicodeInfo.ToUpper(value._value));
```

Comment:
You can ignore the BMP check and write simply `UnsafeCreate(CharUnicodeInfo.ToUpper(value._value))`, just like you did for the lowercase method.

Reply:
will do.